Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Address.m

```diff
@@ -0,0 +1,96 @@
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+#import "BUYClient+Address.h"
+#import "BUYClient+Routing.h"
+#import "BUYClient+Internal.h"
+#import "BUYCustomer.h"
+#import "BUYAddress.h"
+#import "BUYAssert.h"
+
+@implementation BUYClient (Address)
+
+- (BUYRequestOperation *)getAddressesForCustomerID:(NSString *)customerID callback:(BUYDataAddressesBlock)block
+{
+	NSURL *route = [self urlForCustomersAddressesWithID:customerID];
+	return [self getRequestForURL:route completionHandler:^(NSDictionary *json, NSHTTPURLResponse *response, NSError *error) {
+		NSArray<BUYAddress *> *addresses = nil;
+		if (json && !error) {
+			addresses = [self.modelManager insertAddresssWithJSONArray:json[@"addresses"]];
+		}
+		block(addresses, error);
+	}];
+}
+
+- (BUYRequestOperation *)getAddressWithID:(NSNumber *)addressID forCustomerID:(NSString *)customerID callback:(BUYDataAddressBlock)block
+{
+	NSURL *route = [self urlForCustomersAddressWithID:customerID addressID:addressID];
+	return [self getRequestForURL:route completionHandler:^(NSDictionary *json, NSHTTPURLResponse *response, NSError *error) {
+		BUYAddress *address = nil;
+		if (json && !error) {
+			address = [self.modelManager insertAddressWithJSONDictionary:json[@"address"]];
+		}
+		block(address, error);
+	}];
+}
+
+- (BUYRequestOperation *)createAddress:(BUYAddress *)address forCustomerID:(NSString *)customerID callback:(BUYDataAddressBlock)block
+{
+	NSURL *route = [self urlForCustomersAddressesWithID:customerID];
+	return [self postRequestForURL:route object:@{ @"address" : address.JSONDictionary } completionHandler:^(NSDictionary *json, NSHTTPURLResponse *response, NSError *error) {
+		BUYAddress *address = nil;
+		if (json && !error) {
+			address = [self.modelManager insertAddressWithJSONDictionary:json[@"address"]];
+		}
+		block(address, error);
+	}];
+}
+
+- (BUYRequestOperation *)updateAddress:(BUYAddress *)address forCustomerID:(NSString *)customerID callback:(BUYDataAddressBlock)block
+{
+	BUYAssert(address.identifier, @"Failed to update address. Address must have a valid identifier.");
+	
+	NSURL *route = [self urlForCustomersAddressWithID:customerID addressID:address.identifier];
+	return [self putRequestForURL:route object:@{ @"address" : address.JSONDictionary } completionHandler:^(NSDictionary *json, NSHTTPURLResponse *response, NSError *error) {
+		BUYAddress *address = nil;
+		if (json && !error) {
+			address = [self.modelManager insertAddressWithJSONDictionary:json[@"address"]];
+		}
+		block(address, error);
```

Comment:
If the callbacks are non-nullable we should mark them as such, or check for null before calling